src/lib/gssapi/krb5/init_sec_context.c

```diff
@@ -567,6 +567,9 @@ kg_new_connection(
     if (req_flags & GSS_C_DCE_STYLE)
         ctx->gss_flags |= GSS_C_MUTUAL_FLAG;
 
+    /* refuse to set channel bound flag by default */
+    ctx->gss_flags = ctx->gss_flags & (~GSS_C_CHANNEL_BOUND_FLAG);
```

Comment:
As in accept_sec_context, there should be no need to unset channel-bound flag; just don't set it.